New Functions article and related changes. (#3278)
forcedotcom/salesforcedx-vscode: docs/_articles/en/functions/overview.md

---
title: Overview
lang: en
---
## Salesforce Functions (Beta)
Use the Salesforce Functions features in VS Code to build event-driven, elastically scalable apps and experiences. You can create and test functions against sample payloads locally in VS Code.

### Create Function
Run `SFDX: Create Function` to create a Salesforce Function in Javascript or Java in a Salesforce DX project.

This command creates a new directory named after your Function name and all the supporting files.

* VS Code creates the basic scaffolding that contains files with some rudimentary, boilerplate code and supporting metadata.
* The `package.json` file contains information about dependencies.

![Create Function](../../../images/vs_code_create_func.gif)
### Start Function
Run `SFDX: Start Function` to run the function locally.

### Invoke Function
The Salesforce extensions add the `Invoke` and `Debug Invoke` CodeLens to a file that is in the correct format to send test events to a Function that is running locally. With one click, you can test a function that is running, by invoking it with a mock payload. You can also debug the function against the payload.

> A sample `payload.json` file that can invoke a function:

Reviewer: Do we want to explicitly call out that users have to create this payload.json file. There was some confusion around this when I was explaining this to a few users.

---
title: Overview
lang: en
---
## Salesforce Functions (Beta)
Use the Salesforce Functions extensions in VSCode to build event-driven, elastically scalable apps and experiences.

> Use the Salesforce features for VS Code to create and test Salesforce Functions locally.

Reviewer 2: If this sentence is intended to be a lead-in to the head3s, then I would edit it to say something like:
Use these VS Code features to create and test... But I'm not sure this extra sentence adds any real value that the first sentence doesn't already cover.

Author: How is this:
Use the Salesforce Functions features in VSCode to build event-driven, elastically scalable apps and experiences. You can create and test functions against sample payloads locally using the VSCode IDE.  @ekapner

Reviewer 2: @sbudhirajadoc  - one small tweak. 

Use the Salesforce Functions features in VSCode to build event-driven, elastically scalable apps and experiences. You can create and test functions against sample payloads locally in VS Code.

Reviewer 2: I think there's a space between VS and Code - I looked at MS docs and they include a space.

Author: Got it. Thanks Emily!

### Create Function
Run `SFDX: Create Function` to create a Salesforce Function in Javascript or Java in a Salesforce DX project.

> A new directory named after your Function name and all the supporting files are created for you.

Reviewer 2: ```suggestion
This command creates a new directory named after your Function name and all the supporting files.
```

* VS Code creates the basic scaffolding that contains files with some rudimentary, boilerplate code and supporting metadata.
* The `package.json` file contains information about dependencies.

![Create Function](../../../images/vs_code_create_func.gif)
### Start Function

> Run `SFDX: Start Function` from the command palette to run the function locally.

Reviewer 2: You don't say "from the command palette" in Create Function. Do we think that's implied at this point? That all the VS code commands are "from" the palette.

### Invoke Function

> The Salesforce extensions add `Invoke` and `Debug Invoke` CodeLens to a file that is in the correct format to send test events to a Function that is running locally. With one click, you can test a function that is running, by invoking it with a mock payload. You can also debug the function against the payload.

Reviewer 2: CodeLens(es)? or add "the" Invoke Debug Invoke CodeLens...

A sample `payload.json` file that can invoke a function:

![Invoke Debug ](../../../images/vscode_func_payload.png)

### Stop Function

> Stop a function by running `SFDX: Stop Function` from the command palette. This operation simply kills the local process.

Reviewer 2: ```suggestion
Stop a function by running `SFDX: Stop Function`, which simply kills the local process.
```
